lib/iris/fileformats/pp.py

```diff
@@ -1297,6 +1301,16 @@ def core_data(self):
         return self._data
 
     @property
+    def data_dtype(self):
+        return self._data.dtype \
+            if self._data_dtype is None \
+            else self._data_dtype
```

Comment:
> Look closer.

I did look closer! It still took three goes for me to spot the difference even _after_ you assured me there was a difference!

My initial thought on this method was that it is OK to set a temporary variable in some cases, such as where it would require fewer lines if done or improve legibility :wink: